hack/scale_out_poc/two_TP/nginx.conf

```diff
@@ -0,0 +1,144 @@
+    # multi_accept on;
+}
+
+http {
+
+    ##
+    # Basic Settings
+    ##
+
+    sendfile on;
+    tcp_nopush on;
+    tcp_nodelay on;
+    keepalive_timeout 65;
+    types_hash_max_size 2048; 
+    proxy_http_version 1.1; 
+
+    # server_tokens off;
+
+    # server_names_hash_bucket_size 64;
+    # server_name_in_redirect off;
+
+    include /etc/nginx/mime.types;
+    default_type application/octet-stream;
+
+    ##
+    # SSL Settings
+    ##
+
+    ssl_protocols TLSv1 TLSv1.1 TLSv1.2; # Dropping SSLv3, ref: POODLE
+    ssl_prefer_server_ciphers on;
+
+    ##
+    # Logging Settings
+    ##
+
+    access_log /var/log/nginx/access.log;
+    error_log /var/log/nginx/error.log;
+
+        ##
+        # arktos cluster settings
+        #
+
+        server {
+                listen      {{ proxy_port }};
+                server_name {{ proxy_host_name }} {{ proxy_ip }};
+
+                access_log /var/log/nginx/arktos_access.log;
+                error_log /var/log/nginx/arktos_error.log;
+
+                set $resource_api {{ arktos_api_protocol }}://{{ resource_partition_ip }}:{{ arktos_api_port }};
+                set $tenant_api_one {{ arktos_api_protocol }}://{{ tenant_partition_one_ip }}:{{ arktos_api_port }};
+                set $tenant_api_two {{ arktos_api_protocol }}://{{ tenant_partition_two_ip }}:{{ arktos_api_port }};
+
+                #################################################################
+                # forward nodes related requests to resource partition
+                #################################################################
+
+                location ~ ^/api/v1/nodes?(.*) {
+                    proxy_pass $resource_api;
+                }
+
+                location ~ ^/apis/coordination.k8s.io/v1/leases?(.*) {
+                    proxy_pass $resource_api;
+                }
+
+                location ~ ^/apis/([^/])*/([^/])*/tenants/system/namespaces/kube-node-lease/leases?(.*) {
+                    proxy_pass $resource_api;
+                }
+
+                #################################################################
+                # forward non-resource requests back to its source partition
+                #################################################################
+
+                location ~ ^/[a-zA-Z0-9_.-]+$ {
+                    proxy_pass http://$remote_addr:8080;
+                }
+
+                location ~ ^/[a-zA-Z0-9_.-]+/[a-zA-Z0-9_.-]+$ {
+                    proxy_pass http://$remote_addr:8080;
+                }
+
+                location ~ ^/apis/[a-zA-Z0-9_.-]+/[a-zA-Z0-9_.-]+$ {
+                    proxy_pass http://$remote_addr:8080;
+                }
+
+                #################################################################
+                # forward the request of system-tenants or no tenants specificed back to its source partition
+                #################################################################
+
```

Comment:
by using the remote_addr, it assume the request is from the IP the same as the cluster Api Server IP, which is not true, i.e. controller, scheduler, kubelet, or other tools/clients etc can all be on different machine than the Api Server.

Reply:
I see your point. Yet this script is for POC, where each partition is in one box. More work need to be done when different components reside in different machines. Let's just use this script as is for now.